gradle-plugin/src/proguard/gradle/ProGuardTask.java

```diff
@@ -68,37 +77,38 @@
     // (private or not) don't seem to work. Private methods don't work either,
     // but package visible or protected methods are ok.
 
-    @InputFiles
+    @Classpath
     protected FileCollection getInJarFileCollection()
     {
         return getProject().files(inJarFiles);
     }
 
-    @Optional @OutputFiles
+    @OutputFiles
     protected FileCollection getOutJarFileCollection()
     {
-        return getProject().files(outJarFiles);
+        return getObjectFactory().fileCollection().from(getProject().files(outJarFiles).builtBy(this));
```

Comment:
Are the intermediate file collection around outJarFiles and the `builtBy` necessary?  Couldn't this just be `from(outJarFiles)`?  Secondarily, there are places that use `getProject().files()` to create file collections - we should probably be consistent, one way or the other.